File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        
        for i in range(len(nums)):
            num = nums[i]
            diff = target - num
            
            if diff in map:
                return [i, map.get(diff)]
            
            map[num] = i

File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            
            if (map.containsKey(diff) && map.get(diff) != i) {
                return new int[] {map.get(diff), i};
            }
            
            map.put(nums[i], i);
        }
        
        return new int[] {};
    }
}